lib/engine_wrapper.py

```diff
@@ -1149,28 +1150,30 @@ def get_syzygy(board: chess.Board, game: model.Game,
             or chess.popcount(board.occupied) > syzygy_cfg.max_pieces
             or board.uci_variant not in ["chess", "antichess", "atomic"]):
         return None, -3
+
     move: Union[chess.Move, list[chess.Move]]
     move_quality = syzygy_cfg.move_quality
+
     with chess.syzygy.open_tablebase(syzygy_cfg.paths[0]) as tablebase:
         for path in syzygy_cfg.paths[1:]:
             tablebase.add_directory(path)
 
         try:
             moves = score_syzygy_moves(board, dtz_scorer, tablebase)
+            converted_moves = {move: dtz_to_wdl(dtm) for move, dtm in moves.items()}
 
-            best_wdl = max(map(dtz_to_wdl, moves.values()))
-            good_moves = [(move, dtz) for move, dtz in moves.items() if dtz_to_wdl(dtz) == best_wdl]
+            best_wdl = max(converted_moves.values())
+            good_moves = [(move, dtz) for move, dtz in converted_moves.items() if dtz == best_wdl]
```

Comment:
`converted_moves` contains pairs of `(move, wdl)` so `good_moves` contains pairs of `(move, wdl)` instead of `(move, dtz)` leading to the bot choosing a suboptimal move.

Reply:
I made this change because I noticed that `dtz_to_wdl` is applied twice, first to compute the maximum value and second to compute again the value of each move. Since nothing else changes it seems unnecessary.

`converted_moves` is actually made of entries of the type `move: dtz_to_wdl(dtm)`, so the second line should be `(move, wdl) for move, wdl in converted_moves.items() …`

Would that solve the problem or am I wrong?

Reply:
While the variables have incorrect names, renaming them won't help. If you want to avoid recalculation you will have to change `converted_moves = {move: dtz_to_wdl(dtm) for move, dtm in moves.items()}` to `converted_moves = {move: (dtz, dtz_to_wdl(dtz)) for move, dtz in moves.items()}` to store both the dtz and the wdl, and then choose according to what's needed.

Reply:
Mypy raises errors if I use your suggestion. I'm not sure what to do, I could remove the lines we don't agree on so that we can move on with the PR and tackle the issue in another one. You can also directly edit the PR if you want.

Reply:
I agree with removing them for now. The change from copying and pushing to pushing and popping seems to be 20-30% faster, so it would be nice to have it.